src/transformers/models/llava_next/processing_llava_next.py

```diff
@@ -182,13 +183,13 @@ def _get_number_of_features(self, orig_height: int, orig_width: int, height: int
         )
         scale_height, scale_width = height_best_resolution // height, width_best_resolution // width
 
-        patches_height = height // self.patch_size
-        patches_width = width // self.patch_size
+        patches_height = int(math.sqrt(self.num_image_tokens))
+        patches_width = int(math.sqrt(self.num_image_tokens))
         unpadded_features, newline_features = self._get_unpadded_features(
             orig_height, orig_width, patches_height, patches_width, scale_height, scale_width
         )
         # The base patch covers the entire image (+1 for the CLS)
-        base_features = patches_height * patches_width + 1
+        base_features = self.num_image_tokens
```

Comment:
I don't think the numbers quite match here. Even if we assume the patches are square, this also assumes the image is square, which might not be the case. 

Let's say the size of the image is 22x18 and a patch size is 4. 

In the first case: 

    patches_height = 22 // 4 = 5
    patches_width = 18 // 4 = 4

    base_features = patches_height * patches_width + 1
                  = 21 

If we then work backwards

num_image_tokens = 21 
patches_height = patches_width = int(math.sqrt(self.num_image_tokens)) = 4

So this conversion isn't backwards compatible and I think it's also less correct

Reply:
Yes, right. but also I don't think this is possible in available llava-next checkpoints